ci/docker/Dockerfile.build.ubuntu

```diff
@@ -83,9 +82,31 @@ RUN export DEBIAN_FRONTEND=noninteractive && \
         autoconf \
         gperf \
         libb2-dev \
-        libzstd-dev && \
+        libzstd-dev \
+        gfortran && \
     rm -rf /var/lib/apt/lists/*
 
+# Build OpenBLAS from source
+RUN export LIBRARY_PATH=$LIBRARY_PATH:/usr/lib/gcc/x86_64-linux-gnu/7/ && \
+    mkdir ~/openblas && \
+    cd ~/openblas && \
+    OPENBLAS_VERSION=0.3.10 && \
+    wget \
+        https://github.com/xianyi/OpenBLAS/archive/v${OPENBLAS_VERSION}.zip \
+        -O openblas.zip && \
+    unzip -q openblas.zip -d . && \
+    cd OpenBLAS-${OPENBLAS_VERSION} && \
+    CXX="clang++-6.0 -fPIC" CC="clang-6.0 -fPIC" make -j DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 \
+        USE_OPENMP=0 INTERFACE64=1 BINARY=64 && \
+    make PREFIX=/usr/local/opt install && \
```

Comment:
Can we use a more meaningful name than `usr/local/opt`? It will be very confusing to other developers to find two openblas installations. One in /usr/local and one in /usr/local/opt.
For example, use /usr/local/openblas-gcc and /usr/local/openblas-clang?

Reply:
sure I can rename the clang installation to usr/local/openblas-clang. I am a little concerned with renaming the gcc installation because we would need to hard code the blas directory in the many gcc mxnet build whereas we only have few clang builds